### PR TITLE
correction sub-component path in windows when import to ngModule #1719

### PR DESCRIPTION
**Pull request by chgc**
#1719
Correction path in windows
eg: ./parentCompoent\ChildComponet/ChildComponet

### CODE REVIEW

**Comment by chgc** [comment]
I signed it.

**Comment by filipesilva** [comment]
@hansl can you review? This PR is meant to address https://github.com/angular/angular-cli/issues/1719

File: addon/ng2/utilities/route-utils.ts [hunk 1]
```diff
@@ -70,6 +70,9 @@ export function bootstrapItem(mainFile, imports: {[key: string]: [string, boolea
 
 export function insertImport(fileToEdit: string, symbolName: string,
                                   fileName: string, isDefault = false): Change {
+        
+  fileName = fileName.replace(/\\/, '/'); // correction in windows
```
Comment (hansl) [inline addon/ng2/utilities/route-utils.ts]: Please use `path.posix.normalize()` instead of replace. Thanks.

Comment (chgc) [inline addon/ng2/utilities/route-utils.ts]: hi, hansl.
I tried to use path.posix.normalize() . but the result wasn't correct. the Result is shown as below.

before normalize. fileName is './aa\bb/bb.component'
after normalize fileName is 'aa\bb/bb.component'

I couldn't find the reason why this didn't work. my node version is 6.3.1. 

A little more help, please. Thanks

Comment (filipesilva) [inline addon/ng2/utilities/route-utils.ts]: This probably means that the generation of the `'./aa\bb/bb.component'` path is the real issue. Check invocation of `insertImport` and see how the third argument `fileName` is being composed.

Comment (chgc) [inline addon/ng2/utilities/route-utils.ts]: 1. start from blueprint/component/index.js line: 120 

 `
 const componentDir = path.relative(this.dynamicPath.appRoot, this.generatePath);
 const importPath = componentDir ? `./${componentDir}/${fileName}` : `./${fileName}`;
    if (!options['skip-import']) {
      returns.push(
        astUtils.addComponentToModule(modulePath, className, importPath)
          .then(change => change.apply()));
    }
`


2. ast-utils.js line: 291


`
export function addComponentToModule(modulePath: string, classifiedName: string,
    importPath: string): Promise<Change> {
  return _addSymbolToNgModuleMetadata(modulePath, 'declarations', classifiedName, importPath);
}
`

3.in file: ast-utils. _addSymbolToNgModuleMetadata line: 282


`const importInsert: Change = insertImport(ngModulePath, symbolName, importPath);`


the importPath is carried through from the beginning.(blueprint/component and others). 
It seems componentDir is wrong. path.relative(windows environment) will use '\' to return relative path. 


By the way. in route-utils.ts line:205. I used the same trick to fix #1719

**Comment by chgc** [comment]
Closed this pull request  due to ast-utils has been move to separate package. 
this bug remains.

**Pull request closed without merging**